tools/util_generate_hex_from_json.cpp

```diff
@@ -0,0 +1,147 @@
+//   may be used to generate serialization tests cases in other languages and external packages
+//
+
+#include "abieos.h"
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <stdexcept>
+#include <unistd.h>
+#include <vector>
+
+// Main work done here four steps
+//      1) create empty context
+//      2) set the context
+//      3) parse json to binary
+//      4) create hex from bin
+// abiContractName: enum with contract names
+// schema: the name of the data type or reference to schema in the ABI contract
+// json: the name and values
+// verbose: flag to print out step by step messages
+std::string generate_hex_from_json(const char* abi_definition, const char* contract_name, const char* schema, const char* json, bool verbose) {
+    if (verbose) std::cerr << "Schema is: " << schema << " and json is " << json << std::endl << std::endl;
+
+    // create empty context
+    using unique_abieos = std::unique_ptr<abieos_context, decltype( &abieos_destroy )>;
+    unique_abieos context( abieos_create(), &abieos_destroy );
+    if (! context) throw std::runtime_error("unable to create context");
+    if (verbose) std::cerr << "step 1 of 4: created empty ABI context" << std::endl;
+
+    // set the transaction context.
+    // first get the contract_id
+    uint64_t contract_id = abieos_string_to_name(context.get(), contract_name);
+    if (contract_id == 0) {
+        std::cerr << "Error: abieos_string_to_name " << abieos_get_error(context.get()) << std::endl;
+        throw std::runtime_error("unable to set context");
+    }
+    // use our id and set the ABI
+    bool successSettingAbi = abieos_set_abi(context.get(), contract_id, abi_definition);
+    if (! successSettingAbi) {
+        std::cerr << "Error: abieos_set_abi " << abieos_get_error(context.get()) << std::endl;
+        throw std::runtime_error("unable to set context");
+    }
+    if (verbose) std::cerr << "step 2 of 4: established context for transactions, packed transactions, and state history" << std::endl;
+
+    // convert from json to binary. binary stored with context
+    // get contract id returns integer for the ABI contract we passed in by name
+    bool successJsonToBin = abieos_json_to_bin_reorderable(
+        context.get(),
+        contract_id,
+        schema,
+        json
+        );
+    if (!successJsonToBin) {
+        std::cerr << "failed in step 3: using context " << contract_name << std::endl;
+        throw std::runtime_error(abieos_get_error(context.get()));
+    }
+    if (verbose) std::cerr << "step 3 of 4: completed parsing json to binary" << std::endl;
+
+    // now time to return the hex string
+    std::string hex = abieos_get_bin_hex(context.get());
+    if (verbose) std::cerr << "step 4 of 4: converted binary to hex" << std::endl << std::endl;
+    return hex;
+}
+
+// prints usage
+void help(const char* exec_name) {
+    std::cerr << "Usage " << exec_name << ": -f -j JSON -x type [-v]";
```

Comment:
There is a missing `\n` on this line, which mangles the formatting of help output